test/select.jl

```diff
@@ -889,6 +891,93 @@ end
     @test transform(df3, names(df3) .=> (x -> 9) .=> names(df3)) == DataFrame([9 9 9])
     @test transform(df3, names(df3) .=> (x -> 9) .=> names(df3), :x1 => :x4) ==
           DataFrame(ones(0, 4))
+
+    df = DataFrame(x1=1:2, x2=categorical(1:2),
+                   x3=[missing,2], x4=categorical([missing, 2]))
+
+    df2 = select(df, names(df) .=> first)
+    @test df2 ≅ DataFrame(x1_first=1, x2_first=1, x3_first=missing,
+                          x4_first=missing)
+    @test df2.x1_first isa Vector{Int}
+    @test df2.x2_first isa CategoricalVector
```

Comment:
While you're at it, test the element type? That's especially useful for `last` as `Missing` could be dropped.

Reply:
ok - I made tests more specific